include/dos_system.h

```diff
@@ -55,19 +56,37 @@ class DOS_DTA;
 
 class DOS_File {
 public:
-	DOS_File():flags(0),time(0),date(0),attr(0),refCtr(0),open(false),name(0),hdrive(0xff) { };
-	DOS_File(const DOS_File& orig);
-	DOS_File & operator= (const DOS_File & orig);
-	virtual	~DOS_File(){if(name) delete [] name;};
+	DOS_File()
+	        : flags(0),
+	          time(0),
+	          date(0),
+	          attr(0),
+	          refCtr(0),
+	          open(false),
+	          name(""),
+	          hdrive(0xff)
+	{}
+
+	DOS_File(const DOS_File &orig) = default;
+	DOS_File &operator=(const DOS_File &orig);
+
+	virtual ~DOS_File() = default;
+
+	const char *GetName() const { return name.c_str(); }
+
+	void SetName(const char *str) { name = str; }
+
+	bool IsName(const char *str) const
+	{
+		return !name.empty() && (strcasecmp(name.c_str(), str) == 0);
```

Comment:
Could leverage the string's comparing-powers on the left-hand-side of the comparison (make it a bit more readable):

`return str != nullptr && name == str; `

Reply:
I made this in the first version of this patch :) Notice `strcasecmp` instead of `strcmp`. Original code was doing a null-check only `name`, and not on passed parameter. Decided to replace it with empty-check to preserve behaviour in case someone does not set name and then calls `IsName`.